sktime/forecasting/model_selection/_split.py

```diff
@@ -370,10 +370,10 @@ def split_initial(self, y):
             raise ValueError(
                 "Please specify initial window, found: `initial_window`=None"
             )
-
+        fh = self.get_fh()
         initial = check_window_length(self.initial_window)
         initial_training_window = np.arange(initial)
-        initial_test_window = np.arange(initial, len(y))
+        initial_test_window = np.arange(initial, initial + len(fh._values))
```

Comment:
@mloning question regarding this line. I changed this to the length of `fh` and not the full length of `y`. Is this ok? I got an error now because of that, have to see what is going on but would like to have your opinion first on this.

Reply:
Seems ok to me, what does the error say?

Reply:
```python
forecasting\model_selection\_split.py:215: in _split_windows
    end = self._get_end(y)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <sktime.forecasting.model_selection._split.SlidingWindowSplitter object at 0x000002A14CF0DC50>
y = PeriodIndex(['1953-03'], dtype='period[M]', name='Period', freq='M')

    def _get_end(self, y):
        """Helper function to compute the end of the last window"""
        n_timepoints = len(y)
        fh = self._check_fh()
        window_length = check_window_length(self.window_length)

        # end point is end of last window
        is_in_sample = np.all(fh <= 0)
        if is_in_sample:
            end = n_timepoints + 1
        else:
            fh_max = fh[-1]
            end = n_timepoints - fh_max + 1  # non-inclusive end indexing

            # check if computed values are feasible with the provided index
            if window_length is not None:
                if window_length + fh_max > n_timepoints:
                    raise ValueError(
>                       "The window length and forecasting horizon are "
                        "incompatible with the length of `y`"
                    )
E                   ValueError: The window length and forecasting horizon are incompatible with the length of `y`

forecasting\model_selection\_split.py:161: ValueError
=============================================================== warnings summary ===============================================================
c:\users\walmar2\appdata\local\programs\python\python37\lib\site-packages\patsy\constraint.py:13
  c:\users\walmar2\appdata\local\programs\python\python37\lib\site-packages\patsy\constraint.py:13: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    from collections import Mapping

-- Docs: https://docs.pytest.org/en/latest/warnings.html
=========================================================== short test summary info ============================================================
FAILED forecasting\model_selection\tests\test_tune.py::test_gscv_fit[cv2-scoring0-forecaster0-param_dict0] - ValueError: The window length and...
========================================================= 1 failed, 1 warning in 5.70s =========================================================
```
@mloning any idea?

Reply:
```python
forecasting\model_selection\_split.py:215: in _split_windows
    end = self._get_end(y)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <sktime.forecasting.model_selection._split.SlidingWindowSplitter object at 0x000002A14CF0DC50>
y = PeriodIndex(['1953-03'], dtype='period[M]', name='Period', freq='M')

    def _get_end(self, y):
        """Helper function to compute the end of the last window"""
        n_timepoints = len(y)
        fh = self._check_fh()
        window_length = check_window_length(self.window_length)

        # end point is end of last window
        is_in_sample = np.all(fh <= 0)
        if is_in_sample:
            end = n_timepoints + 1
        else:
            fh_max = fh[-1]
            end = n_timepoints - fh_max + 1  # non-inclusive end indexing

            # check if computed values are feasible with the provided index
            if window_length is not None:
                if window_length + fh_max > n_timepoints:
                    raise ValueError(
>                       "The window length and forecasting horizon are "
                        "incompatible with the length of `y`"
                    )
E                   ValueError: The window length and forecasting horizon are incompatible with the length of `y`

forecasting\model_selection\_split.py:161: ValueError
=============================================================== warnings summary ===============================================================
c:\users\...\constraint.py:13
  c:\users\...\constraint.py:13: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    from collections import Mapping

-- Docs: https://docs.pytest.org/en/latest/warnings.html
=========================================================== short test summary info ============================================================
FAILED forecasting\model_selection\tests\test_tune.py::test_gscv_fit[cv2-scoring0-forecaster0-param_dict0] - ValueError: The window length and...
========================================================= 1 failed, 1 warning in 5.70s =========================================================
```
@mloning any idea?

Reply:
Is the window length and forecasting horizon longer than the data (y)?

Reply:
I decided to leave it as it is, now I handle this in `evaluate`. However I think it might be better and more accurate  to change the length of the test data for the `split_initial` to `len(fh._values)`.